components/TransactionForm.js

```diff
@@ -0,0 +1,346 @@
+import { categories } from "@/lib/categories";
+import { useState } from "react";
+import CurrencyInput from "react-currency-input-field";
+import styled from "styled-components";
+
+export default function TransactionForm({ onAdd }) {
+  const today = new Date().toISOString().split("T")[0];
+
+  const [selectedType, setSelectedType] = useState("");
+  const [amount, setAmount] = useState("");
+  const [typeError, setTypeError] = useState(false);
+
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    if (data.amount) {
+      data.amount = parseFloat(
+        data.amount.replace(/\./g, "").replace(",", ".")
+      );
+    }
+
+    if (!selectedType) {
+      setTypeError(true);
+      return;
+    }
```

Comment:
I would put the early return higher up in the function to make the code more efficient.